### DISCUSSION TITLE
callbacks chaining, by returning a pointer to this

### DISCUSSION THREAD

**Discussion by lathoub** [FortySevenEffects/arduino_midi_library#245] [Ideas]
Current
`inline void setHandleMessage(void (*fptr)(const MidiMessage&)) { mMessageCallback = fptr; };`
Suggestion:
`inline MidiInterface& setHandleMessage(void (*fptr)(const MidiMessage&)) { mMessageCallback = fptr; return *this; };`

(add `MidiInterface&` for all callbacks)

This would allow for:
```cpp
MIDI
.setHandleNoteOn([](byte channel, byte note, byte velocity) {
})
.setHandleNoteOff([](byte channel, byte note, byte velocity) {
});
```

Shorter and syntactically _nicer_

A suggestion for a next release.

This has no impact on the use of dynamic memory usage or program storage space.
And no impact on existing code and examples

**Comment by franky47** [comment]
Great idea!

**Comment by lathoub** [comment]
Can also be done for `begin`, `send*` and `turnThruOff` (and all commands returning void)

**Comment by franky47** [comment]
Implemented in #247, thanks @lathoub !
